chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/JHttpEndpoint.java

```diff
@@ -0,0 +1,139 @@
+package com.griddynamics.jagger.invoker.http.v2;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import static com.griddynamics.jagger.invoker.http.v2.JHttpEndpoint.Protocol.HTTP;
+import static com.griddynamics.jagger.invoker.http.v2.JHttpEndpoint.Protocol.HTTPS;
+import static java.lang.String.format;
+
+/**
+ * An object that represents HTTP-endpoint. It consists of {@link JHttpEndpoint#protocol},
+ * {@link JHttpEndpoint#hostname} and {@link JHttpEndpoint#port} fields. <p>
+ *
+ * @author Anton Antonenko
+ * @since 1.3
+ */
+@SuppressWarnings("unused")
+public class JHttpEndpoint {
+
+    /**
+     * Enum representing HTTP and HTTPS protocols
+     */
+    public enum Protocol {
+        HTTP("http"), HTTPS("https");
+
+        private String value;
+
+        Protocol(String value) {
+            this.value = value;
+        }
+    }
+
+    private Protocol protocol = HTTP;
+    private String hostname;
+    private int port = 80;
+
+    public JHttpEndpoint(URI uri) {
+        if (StringUtils.startsWith(uri.getPath().toLowerCase(), HTTP.value))
+            this.protocol = HTTP;
+        else if (StringUtils.startsWith(uri.getPath().toLowerCase(), HTTPS.value))
+            this.protocol = HTTPS;
+        else
+            throw new IllegalArgumentException(format("Protocol of uri '%s' is unsupported!", uri));
+
+        this.hostname = uri.getHost();
+        this.port = uri.getPort();
+    }
+
+    public Protocol getProtocol() {
+        return protocol;
+    }
+
+    public void setProtocol(Protocol protocol) {
+        this.protocol = protocol;
+    }
+
+    public String getHostname() {
+        return hostname;
+    }
+
+    public void setHostname(String hostname) {
+        this.hostname = hostname;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public void setPort(int port) {
+        this.port = port;
+    }
+
+    /**
+     * @return {@link URI} based on {@link JHttpEndpoint#protocol},
+     * {@link JHttpEndpoint#hostname} and {@link JHttpEndpoint#port} fields values.
+     */
+    public URI getURI() {
+        Preconditions.checkNotNull(hostname, "Hostname is null!");
+
+        if (protocol.value == null){
+            protocol = HTTP;
+        }
+        if (port == 80) {
+            return URI.create(format("%s://%s", protocol.value, hostname));
```

Comment:
And do we need all those setters? I think this object should be immutable after creation. Use builder if you need complicated construction. Once it is immutable it is thread safe and we can just cache public URI getURI() value